src/app/hooks/l10n.ts

```diff
@@ -21,17 +21,25 @@ export type GetFragment = (
   id: Parameters<ReactLocalization["getString"]>[0],
   args?: Parameters<ReactLocalization["getElement"]>[2],
   fallback?: Parameters<ReactLocalization["getString"]>[2]
-) => ReturnType<ReactLocalization["getElement"]>;
+) => ReturnType<ReactLocalization["getElement"]> | null;
 
 export type ExtendedReactLocalization = ReactLocalization & {
   getFragment: GetFragment;
 };
 
 export const useL10n = (): ExtendedReactLocalization => {
+  const [isMounted, setIsMounted] = useState(false);
   const { l10n } = useLocalization();
 
-  const getFragment: GetFragment = (id, args, fallback) =>
-    l10n.getElement(createElement(Fragment, null, fallback ?? id), id, args);
+  useEffect(() => {
+    setIsMounted(true);
+  }, [isMounted]);
+
+  const getFragment: GetFragment = (id, args, fallback) => {
+    return isMounted
+      ? l10n.getElement(createElement(Fragment, null, fallback ?? id), id, args)
+      : null;
+  };
```

Comment:
> Wondering what's _hydration_ means in this context.

@flodolo The [hydration errors](https://nextjs.org/docs/messages/react-hydration-error) are specific to React. They are thrown when the HTML content of an element differs between the server and client after React is hydrated (loaded client-side resources).

We have the l10n hook in this file for client-side excecution and [src/app/functions/server/l10n.ts](https://github.com/mozilla/blurts-server/blob/main/src/app/functions/server/l10n.ts) for server-side rendering. Due to how server and client components work since React18 and Next.js 13 client components are being pre-rendered on the server if possible. The client l10n hook will not inject elements on the server though which is resulting in different content between server and client, thus throwing a hydration error which we can avoid with returning `null` at first.